### ISSUE TITLE
rn0.18.0

### ISSUE THREAD

**Issue by liiklin** [axemclion/react-native-cordova-plugin#14]
How to use the rn0.18.0 version?

**Comment by axemclion** [comment]
@liiklin I am looking to create this as an [rnpm](https://github.com/rnpm/rnpm) plugin so that it can be version agnostic.

**Comment by liiklin** [comment]
thx

**Comment by digitalhobo** [comment]
+1

**Comment by matc4** [comment]
+1

**Comment by niftylettuce** [comment]
okay I figured out how to do it, will be submitting a PR shortly

**Comment by axemclion** [comment]
@niftylettuce Awesome - thanks. 

I think we should also look at Cordova 6.0 changes, when have the "permission" changes from Android M.

**Comment by niftylettuce** [comment]
@axemclion I think I am encountering the Cordova 6.0 change issues now... I encounter this error when I try to add it via `node node_modules/.bin/cordova-plugin add cordova-plugin-bluetoothle` per https://github.com/randdusing/cordova-plugin-bluetoothle.

Regarding RN 0.18 instructions:

For RN >= 0.18:

``` diff
+import io.cordova.reactnative.CordovaPluginPackage; // add to top of file
// ...
    @Override
    protected List<ReactPackage> getPackages() {
        return Arrays.<ReactPackage>asList(
-            new MainReactPackage()
+            new MainReactPackage(), // <-- add comma here
+            new CordovaPluginPackage(this, null) // <-- add the package (`null` is for initial bundle/savedInstanceState value)
        );
    }
```

For RN < 0.18:
- Make the class extend _io.cordova.reactnative.CordovaActivity_ instead of _Activity_. For example, replace the line that says `public class MainActivity extends Activity implements DefaultHardwareBa...` with `public class MainActivity extends io.cordova.reactnative.CordovaActivity implements DefaultHardware...`
- Add `.addPackage(super.cordovaPluginPackage)` just below the line `.addPackage(new MainReactPackage())`

**Comment by niftylettuce** [comment]
Sorry, the error I received is the following (I forgot to include in my previous comment):

```log
Users/nexus/Public/Chronos/node_modules/react-native-cordova-plugin/platforms/android/src/com/randdusing/bluetoothle/BluetoothLePlugin.java:583: error: cannot find symbol
    addProperty(returnObj, "requestPermission", cordova.hasPermission(Manifest.permission.ACCESS_COARSE_LOCATION));
                                                       ^
  symbol:   method hasPermission(String)
  location: variable cordova of type CordovaInterface
/Users/nexus/Public/Chronos/node_modules/react-native-cordova-plugin/platforms/android/src/com/randdusing/bluetoothle/BluetoothLePlugin.java:591: error: cannot find symbol
    addProperty(returnObj, "hasPermission", cordova.hasPermission(Manifest.permission.ACCESS_COARSE_LOCATION));
                                                   ^
  symbol:   method hasPermission(String)
  location: variable cordova of type CordovaInterface
/Users/nexus/Public/Chronos/node_modules/react-native-cordova-plugin/platforms/android/src/com/randdusing/bluetoothle/BluetoothLePlugin.java:606: error: cannot find symbol
    cordova.requestPermission(this, REQUEST_ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION);
           ^
  symbol:   method requestPermission(BluetoothLePlugin,int,String)
  location: variable cordova of type CordovaInterface
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
3 errors
:cordovaplugin:compileReleaseJavaWithJavac FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':cordovaplugin:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 13.966 secs
Could not install the app on the device, read the error above for details.
Make sure you have an Android emulator running or a device connected and have
set up your Android development environment:
https://facebook.github.io/react-native/docs/android-setup.html
```

Also, here's what I added to my `AndroidManifest.xml` file:

```diff
+    <uses-permission android:name="android.permission.BLUETOOTH" />
```

**Comment by niftylettuce** [comment]
Sorry, the error I received is the following (I forgot to include in my previous comment):

``` log
Users/nexus/Public/MyProject/node_modules/react-native-cordova-plugin/platforms/android/src/com/randdusing/bluetoothle/BluetoothLePlugin.java:583: error: cannot find symbol
    addProperty(returnObj, "requestPermission", cordova.hasPermission(Manifest.permission.ACCESS_COARSE_LOCATION));
                                                       ^
  symbol:   method hasPermission(String)
  location: variable cordova of type CordovaInterface
/Users/nexus/Public/MyProject/node_modules/react-native-cordova-plugin/platforms/android/src/com/randdusing/bluetoothle/BluetoothLePlugin.java:591: error: cannot find symbol
    addProperty(returnObj, "hasPermission", cordova.hasPermission(Manifest.permission.ACCESS_COARSE_LOCATION));
                                                   ^
  symbol:   method hasPermission(String)
  location: variable cordova of type CordovaInterface
/Users/nexus/Public/MyProject/node_modules/react-native-cordova-plugin/platforms/android/src/com/randdusing/bluetoothle/BluetoothLePlugin.java:606: error: cannot find symbol
    cordova.requestPermission(this, REQUEST_ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION);
           ^
  symbol:   method requestPermission(BluetoothLePlugin,int,String)
  location: variable cordova of type CordovaInterface
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
3 errors
:cordovaplugin:compileReleaseJavaWithJavac FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':cordovaplugin:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 13.966 secs
Could not install the app on the device, read the error above for details.
Make sure you have an Android emulator running or a device connected and have
set up your Android development environment:
https://facebook.github.io/react-native/docs/android-setup.html
```

Also, here's what I added to my `AndroidManifest.xml` file:

``` diff
+    <uses-permission android:name="android.permission.BLUETOOTH" />
```

**Comment by niftylettuce** [comment]
Perhaps the error is related to this http://stackoverflow.com/a/35128940.

**Comment by niftylettuce** [comment]
Okay, for my error, looks like we need to change platformVersion to 5 instead of 4 per https://github.com/axemclion/react-native-cordova-plugin/blob/master/cli/android-cli.js#L52 and https://github.com/axemclion/react-native-cordova-plugin/blob/master/cli/android-cli.js#L65

**Comment by axemclion** [comment]
@niftylettuce This error is due to Cordova 6, where plugins added a new API to check for permissions, something that AndroidM expected

**Comment by niftylettuce** [comment]
@axemclion I'm confused - how do we fix?  What is the issue in code?

**Comment by axemclion** [comment]
@niftylettuce Will take a look at this during the weekend.

**Comment by niftylettuce** [comment]
@axemclion Also I found this for reference, not sure if it helps https://github.com/facebook/react-native/issues/3334

**Comment by niftylettuce** [comment]
Just wanted to give you an update @axemclion that I got it working with my latest work in progress pushed to my fork, if you wanted to take a look!! :+1:

**Comment by axemclion** [comment]
Working on this right now - https://github.com/axemclion/react-native-cordova-plugin/commit/76496c646881f0c875eb2129d8e463b6937cc9a8

I am not going to test it with a couple of plugins, and should publish a new version

**Comment by axemclion** [comment]
@liiklin  I just published 1.0.0 version of this plugin on npm. I also now have an [example app](https://github.com/axemclion/react-native-cordova-plugin/tree/examples) that shows how to use the plugins. 

Closing the issue - if you still find this to be an issue, please re-open it.

**Comment by liiklin** [comment]
@axemclion ok,thank you.
